src/participant/prompts/query.ts

```diff
@@ -0,0 +1,41 @@
+import * as vscode from 'vscode';
+
+import { getHistoryMessages } from './history';
+
+export class QueryPrompt {
+  static getSystemPrompt(): vscode.LanguageModelChatMessage {
+    const prompt = `You are a MongoDB expert.
+Your task is to help the user craft MongoDB queries and aggregation pipelines that perform their task.
+Keep your response concise.
+You should suggest queries that are performant and correct.
+Respond with markdown, suggest code in a Markdown code block that begins with \'\'\'javascript and ends with \`\`\`.
+You can imagine the schema, collection, and database name.
+Respond in MongoDB shell syntax using the \'\'\'javascript code block syntax.`;
+
+    // eslint-disable-next-line new-cap
+    return vscode.LanguageModelChatMessage.Assistant(prompt);
+  }
+
+  static getUserPrompt(
+    request: vscode.ChatRequest
+  ): vscode.LanguageModelChatMessage {
+    // eslint-disable-next-line new-cap
+    return vscode.LanguageModelChatMessage.User(request.prompt);
+  }
+
+  static buildMessages({
+    context,
+    request,
+  }: {
+    request: vscode.ChatRequest;
+    context: vscode.ChatContext;
+  }): vscode.LanguageModelChatMessage[] {
+    const messages = [
+      QueryPrompt.getSystemPrompt(),
+      ...getHistoryMessages({ context }),
```

Comment:
It looks like we don't need to attach the history explicitly to the message. It remembers the conversations within the current open chat.

Reply:
Hmm looks like it can be a bug on the copilot side, because according to documentation the history shouldn't be added automatically https://code.visualstudio.com/api/extension-guides/chat#use-the-chat-message-history